### PR TITLE
[PHI decoupling] remove reference to fluid/framework/lod_tensor.h in phi

### PR DESCRIPTION
**Pull request by RedContritio**
### PR types

Others

### PR changes

Others

### Describe

remove reference to `paddle/fluid/framework/lod_tensor.h` in phi.
- #47615

### CODE REVIEW

**Comment by RedContritio** [comment]
CI 问题待办事项

- [x] GPU 编译测试
- [ ] 分布式编译选项测试

**Comment by RedContritio** [comment]
@YuanRisheng

File: paddle/phi/tests/core/test_custom_kernel.cc [hunk 1]
```diff
@@ -20,14 +20,14 @@ limitations under the License. */
 #include <gtest/gtest.h>
 
 #ifdef _LINUX
-#include "paddle/fluid/framework/lod_tensor.h"
```
Comment (YuanRisheng) [inline paddle/phi/tests/core/test_custom_kernel.cc:23]: 直接删除这个头文件不行吗，为何要把lod_tensor.cc都移过来

Comment (RedContritio) [reply paddle/phi/tests/core/test_custom_kernel.cc:23]: 好，我看看

Comment (RedContritio) [reply paddle/phi/tests/core/test_custom_kernel.cc:23]: 已解决

**Review by luotao1 — COMMENTED** [review]
duplicated in #50475 
2ec1c415469f44fe4d68fa66cff9fcce

**Pull request closed without merging**